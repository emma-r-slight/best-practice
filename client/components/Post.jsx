import React from 'react'

const Post = props => {
  return (
    <li data-testid="post">{props.name}</li>
  )
}

export default Post
