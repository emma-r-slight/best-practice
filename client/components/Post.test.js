import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import '@babel/polyfill'

import Post from './Post'

test('<Post> includes name in <li>', async () => {
  const testName = "test name"
  render(<Post name={testName} />)
  const post = await screen.findByText(testName)
  expect(post).toBeInTheDocument()
  expect(post).toMatchSnapshot()
})
