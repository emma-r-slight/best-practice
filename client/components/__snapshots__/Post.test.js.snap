// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post> includes name in <li> 1`] = `
<li
  data-testid="post"
>
  test name
</li>
`;
